http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/sam/SamServiceInterp.scala

```diff
@@ -34,38 +41,45 @@ class SamServiceInterp[F[_]](apiClientProvider: SamApiClientProvider[F],
     StorageScopes.DEVSTORAGE_READ_ONLY
   )
 
-  // TODO: pattern for retries
-  //  See: https://broadworkbench.atlassian.net/browse/IA-5053
-
-  override def getPetServiceAccount(userInfo: UserInfo, googleProject: GoogleProject)(implicit
+  override def getPetServiceAccount(bearerToken: String, googleProject: GoogleProject)(implicit
     ev: Ask[F, AppContext]
   ): F[WorkbenchEmail] =
     for {
       ctx <- ev.ask
-      googleApi <- apiClientProvider.googleApi(userInfo.accessToken.token)
-      pet <- F.blocking(googleApi.getPetServiceAccount(googleProject.value)).adaptError { case e: ApiException =>
-        SamException.create("Error getting pet service account from Sam", e, ctx.traceId)
+      googleApi <- apiClientProvider.googleApi(bearerToken)
+      userEmail <- getUserEmail(bearerToken)
+      pet <- SamRetry.retry(googleApi.getPetServiceAccount(googleProject.value), "getPetServiceAccount").adaptError {
+        case e: ApiException =>
+          SamException.create("Error getting pet service account from Sam", e, ctx.traceId)
       }
+      _ <- logger.info(ctx.loggingCtx)(
```

Comment:
Added some log statements to these methods.

Note: this class looks a lot like WSM [SamService](https://github.com/DataBiosphere/terra-workspace-manager/blob/main/service/src/main/java/bio/terra/workspace/service/iam/SamService.java), but in Scala. :)